File: src/main/java/spring/webmvc/presentation/controller/AuthController.java
package spring.webmvc.presentation.controller;

import org.springframework.validation.annotation.Validated;
import org.springframework.web.bind.annotation.PostMapping;
import org.springframework.web.bind.annotation.RequestBody;
import org.springframework.web.bind.annotation.RequestMapping;
import org.springframework.web.bind.annotation.RestController;

import lombok.RequiredArgsConstructor;
import spring.webmvc.application.service.AuthService;
import spring.webmvc.presentation.dto.request.MemberLoginRequest;
import spring.webmvc.presentation.dto.request.TokenRequest;
import spring.webmvc.presentation.dto.response.TokenResponse;

@RestController
@RequestMapping("/auth")
@RequiredArgsConstructor
public class AuthController {

	private final AuthService authService;

	@PostMapping("/login")
	public TokenResponse login(@RequestBody @Validated MemberLoginRequest memberLoginRequest) {
		return authService.login(memberLoginRequest);
	}

	@PostMapping("/token/refresh")
	public TokenResponse refreshToken(@RequestBody @Validated TokenRequest tokenRequest) {
		return authService.refreshToken(tokenRequest);
	}
}

File: spring-web-kotlin/src/main/kotlin/spring/web/kotlin/presentation/controller/AuthController.kt
package spring.web.kotlin.presentation.controller

import org.springframework.validation.annotation.Validated
import org.springframework.web.bind.annotation.PostMapping
import org.springframework.web.bind.annotation.RequestBody
import org.springframework.web.bind.annotation.RequestMapping
import org.springframework.web.bind.annotation.RestController
import spring.web.kotlin.application.service.AuthService
import spring.web.kotlin.presentation.dto.request.MemberLoginRequest
import spring.web.kotlin.presentation.dto.request.TokenRequest

@RestController
@RequestMapping("/auth")
class AuthController(
    private val authService: AuthService
) {
    @PostMapping("/login")
    fun login(@RequestBody @Validated memberLoginRequest: MemberLoginRequest) = authService.login(memberLoginRequest)

    @PostMapping("/token/refresh")
    fun refreshToken(@RequestBody @Validated tokenRequest: TokenRequest) = authService.refreshToken(tokenRequest)
}